### PR TITLE
Add Hall Effect Sensor 

### PR DESCRIPTION
**Pull request by brentru**
Component Name: Hall effect sensor - US5881LUA
Component URL: https://www.adafruit.com/product/158

Notes:
* This sensor is "inverted" as it reads 0v (digital low) when a magnetic field is detected, >0V (digital high) otherwise. Description has been updated to reflect this.

### CODE REVIEW

File: components/pin/hall_effect_sensor/definition.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+    "displayName":"Hall Effect Sensor",
+    "mode":"DIGITAL",
+    "direction":"INPUT",
+    "defaultPeriod":60,
+    "selectPullUp": true,
+    "visualization": {
+      "type": "switch",
+      "offLabel": "Magnet Sensed!",
```
Comment (lorennorman) [inline components/pin/hall_effect_sensor/definition.json:9]: FYI: "Magnet Sensed!" will not fit on the switch. Won't break, but will be ugly.

When the user customization is ready you will be able to rapidly test such things before writing the JSON.

**Review by lorennorman — APPROVED** [review]

File: components/pin/hall_effect_sensor/definition.json (hunk 1 above)
Comment (brentru) [reply components/pin/hall_effect_sensor/definition.json:9]: @lorennorman Ok, what's the maximum # of chars that will fit on the switch?

Comment (brentru) [reply components/pin/hall_effect_sensor/definition.json:9]: @lorennorman  bumping above for clarification ^

**Comment by brentru** [comment]
ping @lorennorman  ^ pls see comment above

File: components/pin/hall_effect_sensor/definition.json (hunk 1 above)
Comment (lorennorman) [reply components/pin/hall_effect_sensor/definition.json:9]: Oh hey so sorry I left this hanging, missed a notification!

There's no answer, letters are variable width. What you want to do is open a component viz form on the live site and play with the label inputs with the live preview until you like what you see, then put that here.

**Pull request merged**